lua/apisix/plugins/example-plugin.lua

```diff
@@ -1,13 +1,21 @@
 local core = require("apisix.core")
-
-
--- TODO: need a more powerful way to define the schema
-local args_schema = {
-    i = "int",
-    s = "string",
-    t = "table",
+local rapidjson = require('rapidjson')
+
+
+-- You can follow this document to write schema:
+-- https://github.com/Tencent/rapidjson/blob/master/bin/draft-04/schema
+-- rapidjson not supported `format` in draft-04 yet
+local schema = {
+    type = "object",
+    properties = {
+        i = {type = "number", minimum = 0},
+        s = {type = "string"},
+        t = {type = "array", minItems = 1},
+    },
+    required = {"i"}
 }
 
+local sd = rapidjson.SchemaDocument(schema)
```

Comment:
We can move this action to `core.schema.check_args`, and plugin developers should only care about `schema` and `conf`.